cachi2/core/package_managers/general.py

```diff
@@ -92,30 +92,45 @@ async def _async_download_binary_file(
     log.debug(f"Download completed - {url}")
 
 
+async def on_request_start(
+    session: aiohttp.ClientSession,
+    trace_config_ctx: types.SimpleNamespace,
+    params: aiohttp.TraceRequestStartParams,
+    retry_options: aiohttp_retry.JitterRetry = aiohttp_retry.JitterRetry(
```

Comment:
A suggestion to avoid having the default options here is to use a higher-order function:

```
def _get_on_request_start_function(total_attemps: int) -> Callable:
    async def log_file_download_attempt(
        trace_config_ctx: types.SimpleNamespace,
        params: aiohttp.TraceRequestStartParams,
    ):
        current_attempt = trace_config_ctx.trace_request_ctx["current_attempt"]
        if current_attempt > 1:
            file_name = params.url.path.split("/")[-1]
            log.debug(f"Attempt {current_attempt}/{total_attemps} - {file_name}")

    return log_file_download_attempt
```
(Note that I removed the `session` parameter because it was not being used)

You can use it inside the `async_download_files` function like this:
```
on_request_start = _get_on_request_start_function(num_attempts)
```
This way, the only code change we'd have is moving the `on_request_start` function out of  `async_download_files`, and switching it with a single line to call `_get_on_request_start_function`.

Reply:
I does make me wonder if this effort is worth for covering a few lines of code that do logging :sweat_smile: